config/webpack.config.dev.js

```diff
@@ -179,6 +181,14 @@ module.exports = {
     // a plugin that prints an error when you attempt to do this.
     // See https://github.com/facebookincubator/create-react-app/issues/240
     new CaseSensitivePathsPlugin(),
+    // Warn on using .jsx files, prefer using .js only.
+    // See https://github.com/facebookincubator/create-react-app/issues/290
+    new CheckFilenamePlugin({
+      regex: /\.jsx$/,
```

Comment:
Instead of creating a plugin that can be also solved by a loader (which reuses webpack configuration API):
```
{
  test: /\.jsx$/,
  loader: 'abort',
  query: {
    message: '[resource]: .jsx extensions are not allowed, use .js extension.'
  }
}
```